### ISSUE TITLE
Feature Request - Support for tagging ASM secrets

### ISSUE THREAD

**Issue by alecwcp** [localstack/localstack#3102]
# Type of request: This is a ...

[ ] bug report
[ x ] feature request

# Detailed description
AWS secrets can be tagged (https://docs.aws.amazon.com/cli/latest/reference/secretsmanager/tag-resource.html) and when listed can be filtered by tag (https://docs.aws.amazon.com/cli/latest/reference/secretsmanager/list-secrets.html).

We've recently come across a problem where our solution involves pulling out a list of secrets filtered by tag from AWS, but our local testing differs from deployed code because locally we can create secrets and list them, but not assign tags to them to filter by.

## Expected behavior
- The `tag-resource` subcommand of the `secretsmanager` command will assign tags to secrets.

## Actual behavior
- The `tag-resource` subcommand of the `secretsmanager` throws a `NotImplementedError`.

# Steps to reproduce

**Comment by alecwcp** [comment]
The other addition that would be great at the same time is support for the `filters` option in the `list-secrets` subcommand too. It appears that currently this is just ignored - I was able to run
```
aws --endpoint-url=http://localhost:5601 secretsmanager list-secrets --filters="[{\"Key\":\"tag-key\",\"Values\":[\"MyTagName\"]},{\"Key\":\"tag-value\",\"Values\":[\"Foo\"]}]" --no-paginate
```
(localstack running in docker container exposed on 5601 on the host)
and the secrets that had been created were returned - even though they don't have any tags because the `tag-resource` subcommand isn't supported yet.

**Comment by whummer** [comment]
Thanks for reporting @alecwcp . This should be fixed upstream in spulec/moto#3392 and is now available in LocalStack - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks
